cocos/platform/apple/CCFileUtils-apple.mm

```
@@ -52,18 +52,74 @@ void setBundle(NSBundle* bundle) {
+static void addValueToArray(id item, ValueVector& array);
+static void addObjectToNSArray(const Value& value, NSMutableArray *array);
+
+static id convertCCValueToNSObject(const cocos2d::Value &value)
+{
+    switch (value.getType()) {
+        case Value::Type::NONE:
+            return [NSNull null];
+            
+        case Value::Type::STRING:
+            return [NSString stringWithCString:value.asString().c_str() encoding:NSUTF8StringEncoding];
+            
+        case Value::Type::BYTE:
+            return [NSNumber numberWithInt:value.asByte()];
+            
+        case Value::Type::INTEGER:
+            return [NSNumber numberWithInt:value.asInt()];
+            
+        case Value::Type::UNSIGNED:
+            return [NSNumber numberWithUnsignedInt:value.asUnsignedInt()];
+            
+        case Value::Type::FLOAT:
+            return [NSNumber numberWithFloat:value.asFloat()];
+            
+        case Value::Type::DOUBLE:
+            return [NSNumber numberWithDouble:value.asDouble()];
+            
+        case Value::Type::BOOLEAN:
+            return [NSNumber numberWithBool:value.asBool()];
+            
+        case Value::Type::VECTOR: {
+            NSMutableArray *array = [NSMutableArray array];
+            const ValueVector &vector = value.asValueVector();
+            for (const auto &e : vector) {
+                addObjectToNSArray(e, array);
+            }
+            return array;
+        }
+            
+        case Value::Type::MAP: {
+            NSMutableDictionary *dictionary = [NSMutableDictionary dictionary];
+            const ValueMap &map = value.asValueMap();
+            for (auto iter = map.begin(); iter != map.end(); ++iter) {
+                addObjectToNSDict(iter->first, iter->second, dictionary);
+            }
+            return dictionary;
+        }
+            
+        case Value::Type::INT_KEY_MAP:
+            break;
```

Adding 'default' may hide warning when new Value::Type will appear (it should tell that not all branches are checked). For that case there is default return statement at the end of the function.